### ISSUE TITLE
How to simulate ngChange

### ISSUE THREAD

**Issue by adybionka** [myforce/angularjs-dropdown-multiselect#8]
Hi, 

This directive uses selectedModel instead of standard ngModel. This why ngChange dosn't work. I need to simulate somehow ngChange. I've assigned the same function to onItemSelect, onItemDeselect, onSelectAll and onDeselectAll. But when I click "Select all" onSelectAll event is triggered and also onItemSelect (in the same number as number of options).

So for example, when I want to reload table after select all items, reload method is executed more then once. 

Is it any option to prevent trigger onItemSelect when user clicks "Select all" button? Or is any other possibility to trigger some event when just selectedModel has been changed by dropdown-multiselect directive?

**Comment by pkempenaers** [comment]
If i understand you correctly, you want a simple onSelectionChanged event?

**Comment by adybionka** [comment]
Yes, I want to have one event which will be triggered when user selects single item, all items (by select all button), deselect and deselect all.

**Comment by pkempenaers** [comment]
I've implemented the onSelectionChanged event, It's included from version 1.7.0

**Comment by adybionka** [comment]
Thanks a lot for really quick improvement!

**Comment by adybionka** [comment]
Thanks one more time, but doesn't work like I expected. 

I will describe you what I need by example: We have list with 4 cities. And after select/unselect some city I need to show users from these cities. So onSelectionChanged should be triggered when: 
1) user selects single item - works like expected
2) user unselects single item - works like expected
2) when user clicks select all button, onSelectionChanged should be triggered only once. Now is triggered 5 times. One becuase user click "Select All" button, 4 times because are 4 cities on the list. 
3) when user clicks "Uncheck all" - not onSelectionChanged  is not triggered 

Also I see bug on console:

Error: event.srcElement is undefined
Error: event.srcElement is undefined
.link/$scope.keyDownLink@http://localhost:8080/bower_components/myforce-angularjs-dropdown-multiselect/src/angularjs-dropdown-multiselect.js:344:22
anonymous/fn@http://localhost:8080/bower_components/angular/angular.js line 14432 > Function:2:325
expensiveCheckFn@http://localhost:8080/bower_components/angular/angular.js:15485:18
ngEventHandler/</callback@http://localhost:8080/bower_components/angular/angular.js:25018:17
$RootScopeProvider/this.$get</Scope.prototype.$eval@http://localhost:8080/bower_components/angular/angular.js:17229:16
$RootScopeProvider/this.$get</Scope.prototype.$apply@http://localhost:8080/bower_components/angular/angular.js:17329:20
ngEventHandler/<@http://localhost:8080/bower_components/angular/angular.js:25023:17
defaultHandlerWrapper@http://localhost:8080/bower_components/angular/angular.js:3456:3
createEventHandler/eventHandler@http://localhost:8080/bower_components/angular/angular.js:3444:9

**Comment by pkempenaers** [comment]
Hi,

This should be fixed in version 1.7.1
The other bug in console I couldn't repoduce. Could you make a separate issue for that one?

**Comment by adybionka** [comment]
Thanks!

I will create an issue when I will find way how to reproduce it.

**Comment by pkempenaers** [comment]
Can you also let me know if this now works as expected?

**Comment by adybionka** [comment]
Yes, it works perfectly. Thanks a lot!
